backend/gs/location.go

```diff
@@ -108,18 +110,33 @@ func (l *Location) Exists() (bool, error) {
 
 // NewLocation creates a new location instance relative to the current location's path.
 func (l *Location) NewLocation(relativePath string) (vfs.Location, error) {
-	newLocation := *l
+	if l == nil {
+		return nil, errors.New("non-nil gs.Location pointer is required")
+	}
+
+	//make a copy of the original location first, then ChangeDir, leaving the original location as-is
+	newLocation := &Location{}
+	*newLocation = *l
```

Comment:
As far as I know this is fine, but I definitely don't understand what exactly is happening here. You'll need to go over this with me tomorrow, when I'm in the office.

Reply:
This makes this dereferences the pointer, makes a copy, then returns a pointer to the new copy (and not the original struct).